File: 0075-sort-colors/0075-sort-colors.cpp
class Solution {
public:
    void sortColors(vector<int>& nums) {
        int left=-1,curr=0,right=(int)nums.size()-1;
        while(curr<=right)
        {
            if(nums[curr]==1)
            {
                curr++;
            }
            else if(nums[curr]==0)
            {
                left++;
                swap(nums[curr],nums[left]);
                curr++;
            }
            else
            {
                swap(nums[curr],nums[right]);
                right--;
            }
        }
    }
};

File: 0075-sort-colors/0075-sort-colors.java
class Solution {
    public void sortColors(int[] nums) {
        int[] count = new int[3];
        for(int i=0;i<nums.length;i++)
        {
            count[nums[i]]++;
        }
        int baseIndex=0;
        for(int i=0;i<3;i++)
        {
            for(int j=0;j<count[i];j++)
            {
                nums[baseIndex+j]=i;
            }
            baseIndex+=count[i];
        }
    }
}